controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -192,7 +192,11 @@ func (m *nodeToMachineMapper) Map(object handler.MapObject) []reconcile.Request
 		log.Error(err, "could not get a list of machines")
 	}
 	for _, machine := range machines.Items {
-		if machine.Status.NodeRef != nil && machine.Status.NodeRef.UID == object.Meta.GetUID() {
+		ok := machine.Status.Phase != nil &&
+			len(machine.Status.Addresses) > 0 &&
+			machine.Status.NodeRef != nil &&
+			machine.Status.NodeRef.UID == object.Meta.GetUID()
```

Comment:
Why can't `isValidMachine()` be called here? I understand the signature would have to change, but I am wondering if the main guts of the function applies here.

Reply:
`isValidMachine()` is it is now takes a `runtime.Object` parameter -- the `machine` obj passed here doesn't implement that.